Release/src/http/client/http_client_winhttp.cpp

```diff
@@ -1085,32 +1086,45 @@ class winhttp_client : public _http_client_communicator
+                        for (size_t i = 0; i < pChainContext->cChain; ++i)
                         {
-                            cert_chain.reserve((int)pChainContext->cChain);
-                            for (size_t i = 0; i < pChainContext->cChain; ++i)
+                            auto chain = pChainContext->rgpChain[i];
+                            for (size_t j = 0; j < chain->cElement; ++j)
                             {
-                                auto chain = pChainContext->rgpChain[i];
-                                for (size_t j = 0; j < chain->cElement; ++j)
+                                auto chainElement = chain->rgpElement[j];
+                                auto cert = chainElement->pCertContext;
+                                if (cert)
                                 {
-                                    auto chainElement = chain->rgpElement[j];
-                                    auto cert = chainElement->pCertContext;
-                                    if (cert)
-                                    {
-                                        std::vector<unsigned char> certData(cert->pbCertEncoded, cert->pbCertEncoded + (int)cert->cbCertEncoded);
-                                        cert_chain.emplace_back(certData);
-                                    }
+                                    cert_chain.emplace_back(std::vector<unsigned char>(cert->pbCertEncoded, cert->pbCertEncoded + (int)cert->cbCertEncoded));
                                 }
                             }
                         }
                         CertFreeCertificateChain(pChainContext);
                     }
                     CertFreeCertificateContext(pCert);
                 }
-                if (!p_request_context->m_http_client->client_config().invoke_certificate_chain_callback(url, cert_chain))
+
+                auto info = std::make_shared<certificate_info>(utility::conversions::to_utf8string(web::uri(url).host()), cert_chain, dwErrorStatus);
+
+                if (dwErrorStatus == CERT_TRUST_NO_ERROR || dwErrorStatus == CERT_TRUST_REVOCATION_STATUS_UNKNOWN || dwErrorStatus == (CERT_TRUST_IS_OFFLINE_REVOCATION | CERT_TRUST_REVOCATION_STATUS_UNKNOWN))
+                {
+                    info->verified = true;
+                }
+
+                if (p_request_context->m_http_client->client_config().invoke_certificate_chain_callback(info))
+                {
+                    if (info->verified)
+                    {
+                        p_request_context->m_certificate_chain_verification_failed = false;
+                    }
+                    else
+                    {
+                        p_request_context->m_certificate_chain_verification_failed = true;
+                    }
+                }
+                else
                 {
                     p_request_context->m_certificate_chain_verification_failed = true;
```

Comment:
I know this line is not changed, but if we don't have a callback, then is the verification_failed result not related to the info->verified result?